importer.js

```diff
@@ -92,22 +77,51 @@ function parseFile(file, definitions){
     }
   });
 
-  // Use walkRulesRight: added rules are not parsed.
-  csstree.walkRulesRight(ast, function(rule, item, list) {
+  removeSelectors(ast, definition);
+  return csstree.translate(ast);
+}
+
+function removeSelectors(ast, definition) {
+
+  csstree.walkRules(ast, function (rule, item, list) {
+
+    if (rule.type === 'Atrule') {
+      let atRule = rule;
+      csstree.walkRules(definition, function (defRule) {
+        if (checkAtruleIsSame(atRule, defRule)) {
+          if (defRule.block.children.getSize() === 0) {
+            list.remove(item);
+          }
+        }
+      });
+    }
+
     // Ignore all other types.
     if (rule.type !== 'Rule') {
       return;
     }
 
     // Render selector from ast for comparison.
-    let selector = csstree.translate(rule.prelude)
-    let definition = definitions.find(function (def) {
-      return def.selector === selector;
+    let selector = csstree.translate(rule.prelude);
+    let atRule = this.atrule;
+    let remove = {};
+
+    csstree.walkRules(definition, function (defRule) {
+      let defSelector = csstree.translate(defRule.prelude);
+
+      if (selector === defSelector) {
+        if (checkAtruleIsSame(atRule, this.atrule)) {
+          remove = {
+            selector: selector,
+            declarations: defRule.block.children
+          };
+        }
+      }
     });
 
-    if (definition) {
-      if (definition.declarations.length) {
-        removeDeclarations(rule, item, list, definition.declarations);
+    if (remove.selector) {
+      if (remove.declarations.getSize()) {
```

Comment:
Maybe it would be better to move this into `.walkRules (line 109)` - and then there is no need for `remove` variable and additional checking that it's not empty.